CONTRIBUTING.md

```diff
@@ -0,0 +1,245 @@
+
+Please review these contribution guidelines to ease getting your changes into this repository.
+
+## Style
+
+### Using acronyms
+
+Please avoid acronyms and abbreviations where possible, use them only where the acronym is easier to understand than the long form (example: `SSH` is self-explanatory, `secure shell` is not widely understood).
+
+When you want to use an acronym, please use the long form and the acronym form together at the first use on a page. Example:
+
+> The Ingress Controller (IC) manages incoming traffic to your services.
+
+After that, you can use the acronym without the long form.
+
+### Headline title case
+
+We use [Title Case](https://titlecase.com/) for the main article headline, but not for lower level headlines.
+
+### Code blocks and syntax highlighting
+
+For **code blocks**, we give language hints to ensure proper syntax highlighting. A YAML block, for example, is opened with triple back-ticks followed by `yaml`.
+
+However, shell commands and their output get the fake hint `nohighlight` to prevent any funky syntax highlighting.
+
+Shell commands in code blocks are prepended with a `$ ` (dollar sign and one blank character).
+
+### CLI commands
+
+We use long-form CLI flags and avoid the possible equal sign between flag name and value, for best readability.
+
+Right:
+
+```nohighlight
+gsctl create cluster --owner acme
+```
+
+Wrong:
+
+```nohighlight
+gsctl create cluster --owner=acme
+gsctl create cluster -o=acme
+gsctl create cluster -o acme
+```
+
+Also we break a command into multiple lines once it becomes longer than ~ 60 characters,
+using the backslash character. Example:
+
+```nohighlight
+gsctl create cluster \
+  --owner acme \
+  --create-default-nodepool false
+```
+
+### Linting and validation
+
+Many style rules are checked automatically in CI. You can also execute the check locally
+before pushing commits using the `make lint` command.
+
+For a reference of all rules please check the [DavidAnson/markdownlint documentation](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md).
+
+There is a project specific configuration in place via the `.markdownlint.yaml` file.
+
+To check locally whether all internal links are correct, use `make linkcheck`.
+
+To check both internal and external links, use `make linkcheck-external`.
+
+### Kubernetes API versions
+
+Keep an eye on API versions of kubernetes resources. Examples and references should
+
+- cover all possible API versions of kubernetes versions we provide to our customers
+- if possible, use the latest API versions supported by kubernetes
+- if different kubernetes versions require different API versions, we differentiate which kubernetes version requires which API version by examples.
+
+An example for this would be the deprecation of the deprecation of the API version of Ingress resources from `networking.k8s.io/v1beta1` to `networking.k8s.io/v1` in kubernetes 1.19
```

Comment:
```suggestion
An example for this would be the deprecation of the API version of Ingress resources from `networking.k8s.io/v1beta1` to `networking.k8s.io/v1` in kubernetes 1.19
```